samples/net/https_client/README.rst

```diff
@@ -0,0 +1,123 @@
+.. _https-client-sample:
+
+HTTPS Client
+############
+
+Overview
+********
+
+This sample application shows how to create encrypted HTTP 1.1 requests
+to an HTTPS server and how to parse the incoming responses.
+Supported HTTP 1.1 methods are: GET, HEAD, OPTIONS and POST.
+
+The source code for this sample application can be found at:
+:file:`samples/net/https_client`.
+
+Requirements
+************
+
+- QEMU
+- Terminal emulator software
+- HTTPS Server
+- DNS server (optional)
+
+Building and Running
+********************
+
+Open the project configuration file for your platform, for example:
+:file:`prj_qemu_x86.conf` is the configuration file for QEMU.
+For IPv4 networks, set the following variables:
+
+.. code-block:: console
+
+	CONFIG_NET_IPV4=y
+	CONFIG_NET_IPV6=n
+
+IPv6 is the preferred routing technology for this sample application,
+
+In this sample application, both static IP addresses and DHCPv4 are supported.
+Static IP addresses are specified in the project configuration file,
+for example:
+
+.. code-block:: console
+
+	CONFIG_NET_APP_MY_IPV6_ADDR="2001:db8::1"
+	CONFIG_NET_APP_PEER_IPV6_ADDR="2001:db8::2"
+
+are the IPv6 addresses for the HTTPS client running Zephyr and the
+HTTPS server, respectively. The application also supports DNS resolving so the
+peer address is resolved automatically if host name is given, so you
+can also write the HTTPS server name like this:
+
+.. code-block:: console
+
+	CONFIG_NET_APP_MY_IPV6_ADDR="2001:db8::1"
+	CONFIG_NET_APP_PEER_IPV6_ADDR="6.zephyr.test"
+
+Open the :file:`src/config.h` file and set the server port
+to match the HTTP server setup, for example:
+
+.. code-block:: c
+
+   #define SERVER_PORT		443
+
+assumes that the HTTPS server is listening at the TCP port 443.
+If the default example HTTPS server is used, then the default
+port is 4443.
+
+HTTPS Server
+============
+
+A very simple HTTPS server is provided in net-tool project.
+
+The net-tools can be downloaded from
+
+    https://github.com/zephyrproject-rtos/net-tools
+
+
+Open a terminal window and type:
```

Comment:
Isn't there some configuration you need to do before just starting up the HTTPS server?